bin/BuildPackages.sh

```diff
@@ -21,55 +50,82 @@ set -e
 # an idea what to do for a complete installation of GAP.
 
 
-if [ $# -eq 0 ]
-  then
-    echo "Assuming default GAP location: ../.."
-    GAPDIR=../..
-  else
-    echo "Using GAP location: $1"
-    GAPDIR="$1"
+# Is someone trying to run us from inside the 'bin' directory?
+if [[ -f gapicon.bmp ]]
+then
+  error "This script must be run from inside the pkg directory" \
+        "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# Is someone trying to run us from inside the 'bin' directory?
-if [ -f gapicon.bmp ]
-  then
-    echo "This script must be run from inside the pkg directory"
-    echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
-    exit 1
+# CURDIR
+CURDIR="$(pwd)"
+
+# GAPDIR
+if [[ "$#" -ge 1 ]]
+then
+  case "$1" in
+    --with-gaproot=*)
+      GAPDIR="$(sed 's|--with-gaproot=||' <<< $1)"
```

Comment:
This will break if $1 contains spaces (i.e. if the user specifies a path including spaces). The correct way then is to write this:
```
GAPDIR=$(sed 's|--with-gaproot=||' <<< "$1")
```
Or write this (not using bash extensions):
```
GAPDIR=$(echo "$1 | sed 's|--with-gaproot=||')
```